sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/protobuf/RoslynMetricsImporterTest.java

```diff
@@ -0,0 +1,104 @@
+
+import org.assertj.core.groups.Tuple;
+import org.junit.Before;
+import org.junit.Test;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.FileMetadata;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.issue.NoSonarFilter;
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.api.measures.FileLinesContext;
+import org.sonar.api.measures.FileLinesContextFactory;
+import org.sonar.api.notifications.AnalysisWarnings;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.sonarsource.dotnet.shared.plugins.ProtobufDataImporter.METRICS_FILENAME;
+
+public class RoslynMetricsImporterTest {
+  private static final File TEST_DATA_DIR = new File("src/test/resources/RazorProtobufImporter");
+  private static final String TEST_FILE_PATH = "Cases.razor";
+  private static final File TEST_FILE = new File(TEST_DATA_DIR, TEST_FILE_PATH);
+  private final File protobuf = new File(TEST_DATA_DIR, METRICS_FILENAME);
+
+  @Before
+  public void before() {
+    assertThat(TEST_FILE).withFailMessage("no such file: " + TEST_FILE).isFile();
+    assertThat(protobuf).withFailMessage("no such file: " + protobuf).isFile();
+  }
+
+  @Test
+  public void roslyn_metrics_are_imported() throws FileNotFoundException {
+    var inputFile = new TestInputFileBuilder("dummyKey", TEST_FILE_PATH)
+      .setMetadata(new FileMetadata(mock(AnalysisWarnings.class)).readMetadata(new FileReader(TEST_FILE)))
+      .build();
+
+    var sensorContext = SensorContextTester.create(TEST_FILE);
+    sensorContext.fileSystem().add(inputFile);
+
+    var noSonarFilter = mock(NoSonarFilter.class);
+    var fileLinesContext = mock(FileLinesContext.class);
+    var fileLinesContextFactory = mock(FileLinesContextFactory.class);
+    when(fileLinesContextFactory.createFor(any(InputFile.class))).thenReturn(fileLinesContext);
+
+    new MetricsImporter(sensorContext, fileLinesContextFactory, noSonarFilter, String::toString).accept(protobuf.toPath());
+
+    var measures = sensorContext.measures(inputFile.key());
+    assertThat(measures).hasSize(7);
+
+    assertThat(measures).extracting("metric", "value")
+      .containsOnly(
+        Tuple.tuple(CoreMetrics.CLASSES, 1),
+        Tuple.tuple(CoreMetrics.STATEMENTS, 15),
+        Tuple.tuple(CoreMetrics.FUNCTIONS, 4),
+        Tuple.tuple(CoreMetrics.COMPLEXITY, 4),
+        Tuple.tuple(CoreMetrics.COMMENT_LINES, 0),
+        Tuple.tuple(CoreMetrics.COGNITIVE_COMPLEXITY, 0),
+        Tuple.tuple(CoreMetrics.NCLOC, 12));
+
+    verify(noSonarFilter).noSonarInFile(inputFile, Collections.emptySet());
+
+    // Will be implemented by https://github.com/SonarSource/sonar-dotnet/pull/7915
```

Comment:
Should we prepend a `ToDo` or `FIXME`, to remember not to remove the comment once the PR 7915 is actually implemented?

Reply:
Aren't we supposed to remove it once we merge 7915?

Reply:
The 7915 is currently in progress, @cristian-ambrosini-sonarsource is working on it and will remove it.